pkg/metrics/statsd/e2e/metrics.go

```diff
@@ -0,0 +1,53 @@
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/metrics"
+)
+
+const (
+	DimensionARMGeoLocation = "armGeoLocation"
+	DimensionARMResourceID  = "armResourceID"
+	DimensionResourceType   = "resourceType"
+	DimensionSucceeded      = "Succeeded"
+
+	LogEntryIsE2EEmittableMetric = "IsE2EEmittableMetric"
+	LogEntryMetricName           = "MetricName"
+	LogEntryMetricStatus         = "MetricStatus"
+)
+
+type E2ELogToMetrics struct {
+	m metrics.Emitter
+}
+
+func NewE2ELogToMetrics(m metrics.Emitter) E2ELogToMetrics {
+	return E2ELogToMetrics{
+		m: m,
+	}
+}
+
+func (e *E2ELogToMetrics) PostMetricsFromLogEntry(entry *logrus.Entry) {
+	if v, ok := entry.Data[LogEntryIsE2EEmittableMetric]; ok {
+		if strings.EqualFold(fmt.Sprint(v), "true") {
+			metricName := fmt.Sprint(entry.Data[LogEntryMetricName])
+			metricStatus := fmt.Sprint(entry.Data[LogEntryMetricStatus])
+			metricStatusInt := Btoi(strings.EqualFold(metricStatus, "true"))
+			dimensions := map[string]string{DimensionARMResourceID: fmt.Sprint(entry.Data[DimensionARMResourceID]), DimensionARMGeoLocation: fmt.Sprint(entry.Data[DimensionARMGeoLocation]), DimensionResourceType: fmt.Sprint(entry.Data[DimensionResourceType]), DimensionSucceeded: metricStatus}
+			e.m.EmitGauge(metricName, metricStatusInt, dimensions)
+		}
+	}
+}
+
+func Btoi(b bool) int64 {
```

Comment:
Does this need to be exported?

Reply:
No - is there a more appropriate place in the RP for reusable functions like this?

Reply:
Not that I know of, but it's pretty common to create small functions like this when you need them. Just change it to lowercase so it's not exported.

Reply:
No longer exporting `btoi` function